introduction/idempotency.mdx

```diff
@@ -22,15 +22,16 @@ to be repeated.
 Create actions, like `POST` or `PUT` however require additional gurantees to
 ensure data will not be duplicated. The Invopop APIs handle these scenarios by
 requiring that every model and thus every incoming operation includes an `id`
-field in either UUIDv1 or UUIDv4 string format. If the same create request is
-made twice with the same ID, _and assuming none of the fields have changed_, the
-second request will simply return the same response as the first.
-
-If a create request is made with an existing ID and a different payload, the api
-will respond with a `409 Conflict` response, allowing you to double check what
-might have gone wrong. If the data changes, the request is not idempotent!
+field in UUID[^1] string format. If you send a create request with a previously
+used ID, the API will either return an error or, for certain operations, if
+there are no changes in the payload, the same response as the first request.
```

Comment:
I [tested every single operation in the API](https://docs.google.com/spreadsheets/d/1PRaSVt_aA74Kghc9WtfHRkE9a4vHdmwi3w0FgLg4lPc/edit?usp=sharing) and found that none of the them actually implemented the behaviour described on this page. In summary:

* Currently, most PUT operations succeed the first time a given UUID is used, and fail with a 409 error on every subsequent request _even if the payload is the same_. According to the previous version of this page, if the payload is the same, "the second request will simply return the same response as the first", and that's not what happens in most cases.
* `PUT /sequence/series/:series_id/entries/:id` is one of the exceptions to the previous point. This endpoint always returns 200, regardless of the payload, which seems to be ignored. Closer, but not quite what the docs say.
* The other exception is `PUT /silo/entries/:id`. In this case, if the digest (not the full payload) of the envelope is the same, then subsequent requests with the same ID will return the same response as the first request. If the digest is different, then a 403 error is returned (instead of a 409 one). Amenitiz complained about this particular misinformation.

So I think we need to review and make some fixes to the currently implemented behaviour to make it really consistent across the different endpoints. In the meantime, I suggest we include the change above in the docs (which is necessarily more vague than the previous one, but correct) to avoid misleading users.

@samlown What do you think?

Reply:
Oh! Nice set of tests!

Okay, I do think this should be fixed in the API rather than kicking the bucket down the road as it is extremely useful to to be able to repeat requests and just get the latest data back.

Let's sync on slack to decide quickly which are the most important endpoints to make idempotent.

Reply:
Yep, agreed that we should better fix the API. To move forward with this PR, is it ok that I merge these changes (and change the idempotency page again when the API endpoints are fixed) or do you prefer me to take the changes in idempotency page out of this PR before merging the rest?

Reply:
Let's remove. I'll approve 👍  thanks!!

Reply:
I'll just keep the footnote about UUID and restore the original description here and merge the changes.